kafka-ui-e2e-checks/src/main/java/extensions/QaseExtension.java

```diff
@@ -0,0 +1,137 @@
+
+import java.lang.reflect.Method;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static io.qase.api.QaseClient.getConfig;
+import static io.qase.api.utils.IntegrationUtils.getStacktrace;
+import static org.junit.platform.engine.TestExecutionResult.Status.SUCCESSFUL;
+
+public class QaseExtension implements TestExecutionListener {
+    private static final Logger logger = LoggerFactory.getLogger(QaseExtension.class);
+    private final ApiClient apiClient = QaseClient.getApiClient();
+    private final ResultsApi resultsApi = new ResultsApi(apiClient);
+    private final Map<TestIdentifier, Long> startTime = new ConcurrentHashMap<>();
+
+
+ /*   If you want to run tests with integration Qase.io uncomment static block and set needed parameters
+            (TOKEN, RUN_ID for existing Run or RUN_NAME if you want that new test run will be created in qase.io*/
+
+/*    static {
+        System.setProperty("QASE_ENABLE", "true");
+        System.setProperty("QASE_PROJECT_CODE", "KAFKAUI");
+        System.setProperty("QASE_RUN_ID", "");
+        //   System.setProperty("QASE_RUN_NAME", "Automation run " + new SimpleDateFormat("dd/MM/yyyy HH:mm:ss").format(new Date()));
+        System.setProperty("QASE_API_TOKEN", "");
+        System.setProperty("QASE_USE_BULK", "false");
+    }*/
+
+
+    @Override
+    public void executionStarted(TestIdentifier testIdentifier) {
+        if (QaseClient.isEnabled() && testIdentifier.isTest()) {
+            startTime.put(testIdentifier, System.currentTimeMillis());
+        }
+    }
+
+    @Override
+    public void executionFinished(TestIdentifier testIdentifier, TestExecutionResult testExecutionResult) {
+        if (!QaseClient.isEnabled() || !testIdentifier.isTest()
+                || !startTime.containsKey(testIdentifier)) {
+            return;
+        }
+        Duration duration = Duration.ofMillis(System.currentTimeMillis() - this.startTime.remove(testIdentifier));
+        TestSource testSource = testIdentifier.getSource().orElse(null);
+        Method testMethod = null;
+        if (testSource instanceof MethodSource) {
+            testMethod = getMethod((MethodSource) testSource);
+        }
+        sendResults(testExecutionResult, duration, testMethod);
+    }
+
+    private void sendResults(TestExecutionResult testExecutionResult, Duration timeSpent, Method testMethod) {
+        if (testMethod != null) {
+            ResultCreate resultCreate = getResultItem(testExecutionResult, timeSpent, testMethod);
+
+            try {
+                resultsApi.createResult(getConfig().projectCode(),
+                        getConfig().runId(),
+                        resultCreate);
+                logger.info("Results added to test run with Id = " + getConfig().runId());
+            } catch (QaseException e) {
+                logger.error(e.getMessage());
```

Comment:
```suggestion
                logger.error(e);
```
We don't wanna lose the stacktrace either

Reply:
done